pkg/drivers/kic/kic.go

```diff
@@ -0,0 +1,250 @@
+	return 0, fmt.Errorf("driver does not support GetSSHPort")
+}
+
+// GetURL returns ip of the container running kic control-panel
+func (d *Driver) GetURL() (string, error) {
+	return d.GetIP()
+}
+
+// GetState returns the state that the host is in (running, stopped, etc)
+func (d *Driver) GetState() (state.State, error) {
+	cmd := exec.Command(d.OciBinary, "inspect", "-f", "{{.State.Status}}", d.MachineName)
+	out, err := cmd.CombinedOutput()
+	o := strings.Trim(string(out), "\n")
+	if err != nil {
+		return state.Error, errors.Wrapf(err, "error stop node %s", d.MachineName)
+	}
+	if o == "running" {
+		return state.Running, nil
+	}
+	if o == "exited" {
+		return state.Stopped, nil
+	}
+	if o == "paused" {
+		return state.Paused, nil
+	}
+	if o == "restarting" {
+		return state.Starting, nil
+	}
+	if o == "dead" {
+		return state.Error, nil
+	}
+	return state.None, fmt.Errorf("unknown state")
+
+}
+
+// Kill stops a host forcefully, including any containers that we are managing.
+func (d *Driver) Kill() error {
+	cmd := exec.Command(d.OciBinary, "kill", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "killing kic node %s", d.MachineName)
+	}
+	return nil
+}
+
+// Remove will delete the Kic Node Container
+func (d *Driver) Remove() error {
+	if _, err := d.nodeID(d.MachineName); err != nil {
+		return errors.Wrapf(err, "not found node %s", d.MachineName)
+	}
+	cmd := exec.Command(d.OciBinary, "rm", "-f", "-v", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "error removing node %s", d.MachineName)
+	}
+	return nil
+}
+
+// Restart a host
+func (d *Driver) Restart() error {
+	s, err := d.GetState()
+	if err != nil {
+		return errors.Wrap(err, "get kic state")
+	}
+	if s == state.Paused {
+		return d.Unpause()
+	}
+	if s == state.Stopped {
+		return d.Start()
+	}
+	if s == state.Running {
+		if err = d.Stop(); err != nil {
+			return fmt.Errorf("restarting a running kic node at stop phase %v", err)
+		}
+		if err = d.Start(); err != nil {
+			return fmt.Errorf("restarting a running kic node at start phase %v", err)
+		}
+		return nil
+	}
+
+	// TODO:medyagh handle Stopping/Starting... states
+	return fmt.Errorf("restarted not implemented for kic yet")
+}
+
+// Unpause a kic container
+func (d *Driver) Unpause() error {
+	cmd := exec.Command(d.OciBinary, "pause", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "unpausing %s", d.MachineName)
+	}
+	return nil
+}
+
+// Start a _stopped_ kic container
+// not meant to be used for Create().
+func (d *Driver) Start() error {
+	s, err := d.GetState()
+	if err != nil {
+		return errors.Wrap(err, "get kic state")
+	}
+	if s == state.Stopped {
+		cmd := exec.Command(d.OciBinary, "start", d.MachineName)
+		if err := cmd.Run(); err != nil {
+			return errors.Wrapf(err, "starting a stopped kic node %s", d.MachineName)
+		}
+		return nil
+	}
+	return fmt.Errorf("cant start a not-stopped (%s) kic node", s)
```

Comment:
Would it be helpful to make this function idempotent? As in, if the container is already started, should this just return success?